Gateway/Request/CheckoutDataBuilder.php

```diff
@@ -25,188 +25,236 @@
+
+            $request['dateOfBirth'] = $this->adyenHelper->formatDate($payment->getAdditionalInformation("dob"),
+                'Y-m-d');
+        }
+
+        if ($payment->getAdditionalInformation("telephone")) {
+            $order->getBillingAddress()->setTelephone($payment->getAdditionalInformation("telephone"));
+            $request['telephoneNumber'] = $payment->getAdditionalInformation("telephone");
+        }
+
+        // Additional data for sepa direct debit
+        if ($payment->getAdditionalInformation("ownerName")) {
+            $request['paymentMethod']['sepa.ownerName'] = $payment->getAdditionalInformation("ownerName");
+        }
+
+        if ($payment->getAdditionalInformation("ibanNumber")) {
+            $request['paymentMethod']['sepa.ibanNumber'] = $payment->getAdditionalInformation("ibanNumber");
+        }
+
+        if ($this->adyenHelper->isPaymentMethodOpenInvoiceMethod(
+            $payment->getAdditionalInformation(AdyenHppDataAssignObserver::BRAND_CODE)
+        )) {
+            $openInvoiceFields = $this->getOpenInvoiceData($order);
+            $request = array_merge($request, $openInvoiceFields);
+        }
+
+        //Boleto data
+        if ($payment->getAdditionalInformation("social_security_number")) {
+            $request['socialSecurityNumber'] = $payment->getAdditionalInformation("social_security_number");
+        }
+
+        if ($payment->getAdditionalInformation("firstname")) {
+            $request['shopperName']['firstName'] = $payment->getAdditionalInformation("firstname");
+        }
+
+        if ($payment->getAdditionalInformation("lastName")) {
+            $request['shopperName']['lastName'] = $payment->getAdditionalInformation("lastName");
+        }
+
+        if ($payment->getAdditionalInformation(AdyenBoletoDataAssignObserver::BOLETO_TYPE)) {
+            $boletoTypes = $this->adyenHelper->getAdyenBoletoConfigData('boletotypes');
+            $boletoTypes = explode(',', $boletoTypes);
+
+            if (count($boletoTypes) == 1) {
+                $request['selectedBrand'] = $boletoTypes[0];
+                $request['paymentMethod']['type'] = $boletoTypes[0];
+            } else {
+                $request['selectedBrand'] = $payment->getAdditionalInformation("boleto_type");
+                $request['paymentMethod']['type'] = $payment->getAdditionalInformation("boleto_type");
+            }
+
+            $deliveryDays = (int)$this->adyenHelper->getAdyenBoletoConfigData("delivery_days", $storeId);
+            $deliveryDays = (!empty($deliveryDays)) ? $deliveryDays : 5;
+            $deliveryDate = date(
+                "Y-m-d\TH:i:s ",
+                mktime(
+                    date("H"),
+                    date("i"),
+                    date("s"),
+                    date("m"),
+                    date("j") + $deliveryDays,
+                    date("Y")
+                )
+            );
+
+            $request['deliveryDate'] = $deliveryDate;
+        }
+
```

Comment:
better to move to the boletoDataBuilder as this is still a seperate payment method or not ?

Reply:
I think we can put it in AdyenPaymentBoletoAuthorizeRequest as we load this before for boleto specifics.

Reply:
we removed the specific boletobuilder, do we really need it?